modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupBuilder.java

```diff
@@ -374,6 +375,14 @@ public void addPainlessClass(Class<?> clazz, Map<Class<?>, Object> annotations)
                     }
 
                     canonicalClassNamesToClasses.put(importedCanonicalClassName.intern(), clazz);
+                    if (annotations.get(AliasAnnotation.class)instanceof AliasAnnotation alias) {
+                        Class<?> existing = canonicalClassNamesToClasses.put(alias.alias(), clazz);
+                        if (existing != null) {
+                            throw new IllegalArgumentException(
+                                "Cannot have same alias [" + alias.alias() + "] for multiple classes [" + existing + "," + clazz + "]"
```

Comment:
I think this isn't just about alias, it can shadow existing class names right? Maybe tweak the wording to make clear it's either an existing alias or classname

Reply:
Indeed, reworded.